environment/src/test/java/com/sequenceiq/environment/environment/validation/network/azure/AzurePrivateEndpointValidatorTest.java

```diff
@@ -229,29 +231,60 @@ void testCheckExistingPrivateDnsZoneWhenNotPrivateEndpoint() {
     }
 
     @Test
-    void testCheckExistingPrivateDnsZoneWhenPrivateEndpoint() {
+    void testCheckExistingServicePrivateDnsZoneWhenPrivateEndpoint() {
+        ValidationResult.ValidationResultBuilder validationResultBuilder = new ValidationResult.ValidationResultBuilder();
+        AzureParams azureParams = getAzureParams(EXISTING_DATABASE_PRIVATE_DNS_ZONE_ID, null);
+        NetworkDto networkDto = getNetworkDto(azureParams);
+        when(azureExistingPrivateDnsZonesService.hasNoExistingServiceZones(networkDto)).thenReturn(false);
+
+        underTest.checkExistingServicePrivateDnsZone(validationResultBuilder, new EnvironmentDto(), networkDto);
+
+        assertFalse(validationResultBuilder.build().hasError());
+        verify(credentialToCloudCredentialConverter).convert(any());
+        verify(azureClientService).getClient(any());
+        verify(azureExistingPrivateDnsZoneValidatorService).validate(any(), any(), any(), any(), eq(validationResultBuilder));
+    }
+
+    @Test
+    void testCheckExistingRegisteredOnlyPrivateDnsZoneWhenNoRegisteredOnlyZones() {
+        ValidationResult.ValidationResultBuilder validationResultBuilder = new ValidationResult.ValidationResultBuilder();
+        AzureParams azureParams = getAzureParams(null, null);
+        NetworkDto networkDto = getNetworkDto(azureParams);
+        when(azureExistingPrivateDnsZonesService.hasNoExistingRegisteredOnlyZones(networkDto)).thenReturn(true);
+
+        underTest.checkExistingRegisteredOnlyPrivateDnsZone(validationResultBuilder, new EnvironmentDto(), networkDto);
+
+        assertFalse(validationResultBuilder.build().hasError());
+        verify(credentialToCloudCredentialConverter, never()).convert(any());
+        verify(azureClientService, never()).getClient(any());
+        verify(azureExistingPrivateDnsZoneValidatorService, never()).validate(any(), any(), any(), any(), any());
+    }
+
+    @Test
+    void testCheckExistingRegisteredOnlyPrivateDnsZone() {
```

Comment:
the test name would be nicer if the when criteria also contains what is the expected result here